### PR TITLE
NyanoCombat Part 4: MeleeSystem Refactor Part 2, Electric Boogaloo

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

Surprise motherfucker, here's a massive fucking refactor of SharedMeleeWeaponSystem. 
The MeleeWeaponComponent now allows for melee weapons to directly configure how and what members of the ContestsSystem they wish to utilize for influencing damage. Additionally, the damage of a melee weapon modified by Contests is now also visible when examined(Although this only reflects damage as per the current condition of the user). 

# Changelog

:cl:
- add: Melee Weapons can now individually define their interactions with the ContestsSystem.
- add: Added the ContestArgs type, allowing arbitrarily any component to contain a list of arguments for ContestSystems.
- add: Added the ContestConstructor, a new type of meta-Contest that enables other systems to use components to define all possible contest behaviors, rather than needing to hardcode specific interactions.

### CODE REVIEW

**Review by OldDanceJacket — APPROVED** [review]

File: Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs [hunk 1]
```diff
@@ -49,6 +50,7 @@ public abstract class SharedMeleeWeaponSystem : EntitySystem
     [Dependency] private readonly IPrototypeManager _protoManager = default!;
     [Dependency] private readonly StaminaSystem _stamina = default!;
     [Dependency] private readonly ContestsSystem _contests = default!;
+    [Dependency] private readonly IConfigurationManager _config = default!;
```
Comment (FoxxoTrystan) [inline Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs:53]: Why is this added if not used, just asking.

Comment (VMSolidus) [reply Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs:53]: Probably a leftover from when the Contest Constructor lived in a Partial of SharedMeleeWeaponSystem.
